adapters/repos/db/vector/hnsw/search.go

```diff
@@ -179,6 +186,8 @@ func (h *hnsw) searchLayerByVectorWithDistancer(queryVector []float32,
 		floatDistancer = h.distancerProvider.New(queryVector)
 	}
 
+	entryPointElem := entrypoints.Pop()
+	entrypoints.Insert(entryPointElem.ID, entryPointElem.Dist)
```

Comment:
**question:** why the next entrypoint is popped and then inserted again?

Reply:
Nice catch... was a test I did but never removed the code...